kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -64,44 +69,76 @@ class KyuubiSessionImpl(
   private var _handle: SessionHandle = _
   override def handle: SessionHandle = _handle
 
+  @volatile
+  private var _initEngineOpHandle: OperationHandle = _
+  def initEngineOpHandle: OperationHandle = _initEngineOpHandle
+
   override def open(): Unit = {
     MetricsSystem.tracing { ms =>
       ms.incCount(CONN_TOTAL)
       ms.incCount(MetricRegistry.name(CONN_OPEN, user))
     }
-    withZkClient(sessionConf) { zkClient =>
-      val (host, port) = engine.getOrCreate(zkClient)
-      openSession(host, port)
-    }
+    _handle = SessionHandle(protocol)
```

Comment:
this seems a breaking change? @ulysses-you ?

Reply:
yeah, we should keep it as null until we opened the enign session. so we can use engien session handle as kyuubi server side session handle.

Reply:
> yeah, we should keep it as null until we opened the enign session. so we can use engien session handle as kyuubi server side session handle.

this will not work as in this feature, the session handle shall be returned ASAP

Reply:
in this case we can turn it back to a `val` to keep the old way

Reply:
if so , we may need reocrd two session handle in event log